docs/proposals/106-spot-instance-karpenter/README.md

```diff
@@ -0,0 +1,450 @@
+this Proposal.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a Proposal within the wider InftyAI community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+A `llama2-7B` model can be running on __1xA100__ GPU, also on __1xA10__ GPU, even on __1x4090__ and a variety of other types of GPUs as well, that's what we called resource fungibility. In practical scenarios, we may have a heterogeneous cluster with different GPU types, and high-end GPUs will stock out a lot, to meet the SLOs of the service as well as the cost, we need to schedule the workloads on different GPU types. With the [ResourceFungibility](https://github.com/InftyAI/scheduler-plugins/blob/main/pkg/plugins/resource_fungibility) in the llmaz-scheduler, we can simply achieve this with at most 8 alternative GPU types.
+
+To optimize cost, users commonly deploy their cluster in the cloud and rely on dynamic node scaling to match resource allocation with workload demand. This elasticity helps handle traffic spikes efficiently while avoiding overprovisioning and idle resource waste. In such setups, Karpenter is widely used as the provisioning engine, automatically launching the right instance types — including GPU-backed instances — in response to pending pods.
+
+However, Karpenter is built to adhere to the scheduling decisions of kube-scheduler. So it's certainly possible we would run across some cases where Karpenter makes incorrect decisions when a custom scheduler is in the mix. As a result, it may launch a cheaper node with __1xT4__ GPU which is incompatible with the associated model's inference configuration, leading to pods remaining in the Pending state despite available resources in node pools. The root cause is that Karpenter is not aware of the llmaz-scheduler's scheduling constraints. 
+
+### Goals
+
+<!--
+List the specific goals of the Proposal. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Provision spot instances for inference workloads based on the model's flavor requirements.
+- Support flexible and preferred GPU scheduling with spot instances.
+- This proposal is only for AWS but the implementation can be extended to other cloud providers.
+
+### Non-Goals
+
+<!--
+What is out of scope for this Proposal? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Integration with the [Kubernetes Cluster Autoscaler](https://github.com/kubernetes/autoscaler) is out of scope for this proposal.
+- Add custom scheduler support for the upstream of the Karpenter project, and it is tracked in [this issue](https://github.com/kubernetes-sigs/karpenter/issues/742). Once the support is added in the upstream, we don't need to maintain the [forked version](https://github.com/InftyAI/karpenter).
```

Comment:
Add a non-goal for multi-host inference here:
- Support for multi-host inference in out-of-the-scope now because reclaiming one part of the service requires reclaiming other parts as well which brings instability to the service.